docs(core): improve viewEncapsulation documentation (#44151)
angular/angular: aio/content/guide/view-encapsulation.md

- `None` means that Angular does no view encapsulation.
  Angular adds the CSS to the global styles.
  The scoping rules, isolations, and protections discussed earlier don't apply.
  This mode is essentially the same as pasting the component's styles into the HTML.
- `ViewEncapsulation.Emulated`, it instructs Angular to programmatically modify the component's CSS selectors so that they are only applied to the component's view and do not affect other elements in the application (_emulating_ a ShadowDom-like behavior). For more details, see [Inspecting generated CSS](guide/view-encapsulation#inspect-generated-css).

To set the component's encapsulation mode, use the `encapsulation` property in the component metadata:
- `ViewEncapsulation.None`, it doesn't instruct Angular to apply any sort of view encapsulation.
  Meaning that any CSS styles specified for the component are actually globally applied and can affect and HTML element present within the application.
  This mode is essentially the same as including the styles into the HTML itself.

> <code-example path="component-styles/src/app/quest-summary.component.ts" region="encapsulation.shadow" header="src/app/quest-summary.component.ts"></code-example>

Author: I've removed the example as it didn't seem beneficial to me as:
 - after my changes the introductory text above already makes clear how to apply the strategy so I don't really think actually showing it adds much here
 - in the same docs page (below) there are various examples showing the application of the encapsulation strategy, so this example again didn't seem to add much to me

> <div class="alert is-important">

Author: maybe it should be `is-helpful`? :man_shrugging:

Reviewer: I think this is fine unless @jelbourn (or anyone else has a view).

>   Not all browsers support it, which is why the `ViewEncapsulation.Emulated` is the recommended and default mode.

Author: I changed "The support is still limited" to "Not all browsers support it" because as you can see in the [caniuse link](https://caniuse.com/shadowdomv1) the support is actually pretty good I think

}
</code-example>

These styles are post-processed so that each CSS selector is augmented with the appropriate `_nghost` or `_ngcontent` attribute. These modified selectors make sure the styles to be applied to components' views in an isolated and targeted fashion.

## Mixing encapsulation modes

> As previously mentioned the encapsulation mode can be specified in the Component's decorator on a _per component_ basis, this means that within your application you can have different components using different encapsulation strategies.

Reviewer: ```suggestion
As previously mentioned you specify the encapsulation mode in the Component's decorator on a _per component_ basis, this means that within your application you can have different components using different encapsulation strategies.
```

> Although possible, this is not recommended, in case it is really necessary for your needs you should be aware of how the styles of components using different encapsulation modes would interact with each other:

Reviewer: ```suggestion
Although possible, this is not recommended. If it is really needed you should be aware of how the styles of components using different encapsulation modes will interact with each other:
```

> - The styles of components with `ViewEncapsulation.Emulated` are added to the `<head>` of the document, making them available throughout the application, but their selectors are modified as illustrated in the previous section, meaning that they only affect elements within the respective components' templates.

Reviewer: I think this can be simplified, since we explain the mechanism earlier.

```suggestion
- The styles of components with `ViewEncapsulation.Emulated` are added to the `<head>` of the document, making them available throughout the application, but their selectors only affect elements within their respective components' templates.
```

> - The styles of components with `ViewEncapsulation.None` are added to the `<head>` of the document, making them available throughout the application, they are not modified in any way making them completely global and applicable to any element withing the application's DOM.

Reviewer: ```suggestion
- The styles of components with `ViewEncapsulation.None` are added to the `<head>` of the document, making them available throughout the application, so are completely global and affect any matching elements within the document.
```

> - The styles of components with `ViewEncapsulation.ShadowDom` are only added to the shadow DOM host, ensuring that they only affect elements within the components' views.

Reviewer: ```suggestion
- The styles of components with `ViewEncapsulation.ShadowDom` are only added to the shadow DOM host, ensuring that they only affect elements within their respective components' views.
```

#### No encapsulation

> This first example shows a component that has `ViewEncapsulation.None`. This component colors its template elements red.

Reviewer: I am not sure that replacing `The` with `This` is an improvement... 🤔

Author: It sounds more clear to me :sweat_smile: , but if it's not I'll just revert this :slightly_smiling_face::+1:

This first example shows a component that has `ViewEncapsulation.None`. This component colors its template elements red.

<code-example path="view-encapsulation/src/app/no-encapsulation.component.ts" header="src/app/no-encapsulation.component.ts"></code-example>>

Angular adds the styles for this component as global styles to the `<head>` of the document.

> **Angular also adds the styles to all shadow DOM hosts.** Therefore, the styles are available throughout the application.

Author: this was already mentioned above so I don't think there is a need to emphasise it

> As mentioned Angular also adds the styles to all shadow DOM hosts. Therefore, the styles are available throughout the whole application.

Reviewer: ```suggestion
As already mentioned Angular also adds the styles to all shadow DOM hosts. Therefore, the styles are available throughout the whole application.
```

Author: my bad sorry :sweat::+1:

<img src="generated/images/guide/view-encapsulation/no-encapsulation.png" alt="component with no encapsulation">

#### Emulated encapsulation

This second example shows a component that has `ViewEncapsulation.Emulated`. This component colors its template elements green.

<code-example path="view-encapsulation/src/app/emulated-encapsulation.component.ts" header="src/app/emulated-encapsulation.component.ts"></code-example>>

> Similar to `ViewEncapsulation.None`, Angular adds the styles for this component to the `<head>` of the document, and to all the shadow DOM hosts (but in this case, the styles are not applied to Shadow DOM elements since they are "scoped" via extra attributes as described in ["Inspecting generated CSS"](#inspecting-generated-css)).

Reviewer: The second example does not demonstrate any Shadow DOM so I think the second half of this sentence is not needed here.

Author: yeah good call, it feels a bit out of place, sorry for not noticing it :sweat:
